build-packages/scandipwa-template-plugin-system/util/ensure-valid-value.js

```diff
@@ -0,0 +1,31 @@
+const { STRING_API_KEY, DOM_API_KEY } = require('./exposed-api-keys');
+
+const PluginInvalidDomException = require('../errors/plugin-invalid-dom');
+const PluginInvalidStringException = require('../errors/plugin-invalid-string');
+
+const exceptionMap = {
+    [DOM_API_KEY]: PluginInvalidDomException,
+    [STRING_API_KEY]: PluginInvalidStringException
+};
+
+const faultyCaseMap = (value) => ({
+    [DOM_API_KEY]: [
+        !value,
+        typeof value !== 'object',
+        Array.isArray(value)
+    ],
+    [STRING_API_KEY]: [
+        !value,
+        typeof value !== 'string'
+    ]
+});
```

Comment:
Instead of two maps, create one bump with an object as value. It will be much easier to read. Otherwise your two object properties may become un-sorted or objects too big thus loosing the easy readability property they have now.

Reply:
Refactored. Check out the new implementation, I hope I got your thought correctly :)